migrations/lib/importer/steps/tags.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+module Migrations::Importer::Steps
+  class Tags < ::Migrations::Importer::CopyStep
+    include ::HasSanitizableFields
+
+    MAX_DESCRIPTION_LENGTH = 1000
+    RESERVED_TAGS = Tag::RESERVED_TAGS.to_set.freeze
+
+    store_mapped_ids true
+    requires_mapping :existing_tag_by_name, "SELECT LOWER(name), id FROM tags"
+
+    column_names %i[id name description target_tag_id created_at updated_at]
+
+    total_rows_query <<~SQL, MappingType::TAGS
+      SELECT COUNT(*)
+      FROM tags
+           LEFT JOIN mapped.ids mapped_tag
+            ON tags.original_id = mapped_tag.original_id AND mapped_tag.type = ?
+      WHERE mapped_tag.original_id IS NULL
+    SQL
+
+    rows_query <<~SQL, MappingType::TAGS
+      SELECT tags.*
+      FROM tags
+          LEFT JOIN mapped.ids mapped_tag
+            ON tags.original_id = mapped_tag.original_id AND mapped_tag.type = ?
+      WHERE mapped_tag.original_id IS NULL
+      ORDER BY tags.target_tag_id, tags.ROWID
+    SQL
+
+    def execute
+      SiteSetting.max_tag_length = 100 if SiteSetting.max_tag_length < 100
```

Comment:
I copied this over from the current importer as-is, but I’m not sure why the value is set to 100.

Should we wrap site setting updates like this in some kind of restore logic, or maybe have a dedicated, configurable step at the start of the import to set them with a warning? We could potentially merge these with the site settings specified in the intermediate DB.

Reply:
I do not know why it's 100. My best guess is that it was an easy way out during the migration I was working on. `max_tag_length` needs to be larger than the longest tag name. Looking at the implementation of `DiscourseTagging.clean_tag` it seems like tag names can't get longer, just shorter. So, maybe we simply do a `LENGTH(name)` query on the IntermediateDB first, and then adjust the site setting if the value is too small. What do you think about that?

Reply:
I’m okay with adjusting it based on the longest tag in the current import.

The bigger concern here though is how we handle site setting changes like this during import runs. `max_tag_length` change here isn’t a big deal on its own, but other settings might have more impact, especially when importing or merging into existing sites. 

Once we have logging or warnings in place, maybe we start flagging changes like these? Also, should this be handled in the dedicated site settings import step, where we collate all changes needed and present  a warning upfront?

Reply:
I think it's best to keep it within the step that requires the change. Makes the code easier to maintain. But, we could put those setting changes in a dedicated method of steps and call them at the beginning of the import.